<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DemoLR1Window</class>
 <widget class="QMainWindow" name="DemoLR1Window">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>738</width>
    <height>492</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>LR(0)/SLR 自动机演示</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="HorizontalScrollArea" name="scrollArea">
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="scrollAreaWidgetContents">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>724</width>
         <height>59</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QCommandLinkButton" name="step2">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>转换增广文法</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCommandLinkButton" name="step3">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>计算First集</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCommandLinkButton" name="step5">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>计算项目集闭包</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCommandLinkButton" name="step6">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>生成自动机</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QWidget" name="widget" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>1</verstretch>
       </sizepolicy>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <property name="spacing">
        <number>0</number>
       </property>
       <item row="0" column="0">
        <widget class="QSplitter" name="splitter">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <widget class="QWidget" name="widget_2" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>1</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QToolButton" name="stepButton">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>单步执行</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QToolButton" name="runButton">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>运行到下一算法</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QToolButton" name="runToCursorButton">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>运行到选中步骤</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QListWidget" name="listWidget">
             <property name="editTriggers">
              <set>QAbstractItemView::NoEditTriggers</set>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="LR1VariableWidget" name="keyWidget" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>3</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="font">
           <font>
            <pointsize>14</pointsize>
           </font>
          </property>
          <property name="cursor">
           <cursorShape>OpenHandCursor</cursorShape>
          </property>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar">
   <property name="styleSheet">
    <string notr="true">QStatusBar::item{border: 0px}</string>
   </property>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>HorizontalScrollArea</class>
   <extends>QScrollArea</extends>
   <header>view/HorizontalScrollArea.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>LR1VariableWidget</class>
   <extends>QWidget</extends>
   <header>widget/LR1VariableWidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
